### PR TITLE
vesting: Ensure claimable() only returns for valid schedules

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: contracts/token/vesting/ERC20Vesting.sol [hunk 1]
```diff
@@ -99,7 +99,9 @@ contract ERC20Vesting is IERC20Vesting {
     }
 
     function claimable(address receiver) external view override returns (uint256) {
-        return _claimable(vestingTerms[receiver]);
+        VestingTerms memory terms = vestingTerms[receiver];
+        require(isScheduleValid(terms), "No vesting data for receiver.");
```
Comment (yuval-weiss) [inline contracts/token/vesting/ERC20Vesting.sol:103]: Can we move the require statement to the internal `_claimable` function? The other consumer of `_claimable` (`claim()`) also checks `isScheduleValid`...

**Comment by axic** [comment]
@mijovic this is still a question.

**Comment by mijovic** [comment]
> @mijovic this is still a question.

Well if we want checks to be in place, we should try including this PR as well.
Maybe moving check to _claimable as @yuval-weiss suggested?

**Comment by axic** [comment]
> Maybe moving check to `_claimable` as @yuval-weiss suggested?

I'm not sure actually, because most of the functions need this check, even those which are not using `_claimable`. So in that sense perhaps it is more clear to have it in the function body as opposed to hidden in a helper.

File: contracts/token/vesting/ERC20Vesting.sol [hunk 2]
```diff
@@ -113,7 +113,9 @@ contract ERC20Vesting is IERC20Vesting {
     }
 
     function claimable(address receiver) external view override returns (uint256) {
```
Comment (axic) [inline contracts/token/vesting/ERC20Vesting.sol:115]: Alternative is if we just add a note to natspec saying this will return 0 for non-existent accounts.

Comment (mijovic) [reply contracts/token/vesting/ERC20Vesting.sol:115]: this is also fine by me

**Review by mijovic — APPROVED** [review]

**Pull request merged**